### PR TITLE
Adding EnableSecureBoot functionality

### PR DESCRIPTION
**Pull request by TSKushal**
SUMMARY
Adding functionality to enable SecureBoot on a server.

ISSUE TYPE
- Feature Pull Request

COMPONENT NAME
redfish_config.py
redfish_utils.py

### CODE REVIEW

File: plugins/module_utils/redfish_utils.py [hunk 1]
```diff
@@ -3152,3 +3152,53 @@ def set_session_service(self, sessions_config):
...
+                "ret": True,
+                "changed": False,
+                "msg": "Secure boot is already enabled"
+            }
+        else:
+            # Check server power state
+            state = server_details["PowerState"]
+
+            # If server is not powered OFF, then failing with message
+            if state != "Off":
```
Comment (mraineri) [inline plugins/module_utils/redfish_utils.py:3185]: This isn't necessarily something required by everyone. Services can support modifying this property while the system is on.

File: plugins/module_utils/redfish_utils.py [hunk 2]
```diff
@@ -3152,3 +3152,53 @@ def set_session_service(self, sessions_config):
...
+
+            body = {}
+            body["SecureBootEnable"] = True
+
+            res = self.patch_request(self.root_uri + secure_boot_url, body)
+
+            if not res["ret"]:
+                return res
+
+            return {
```
Comment (mraineri) [inline plugins/module_utils/redfish_utils.py:3200]: With the changes made prior to supporting payload inspection with the `patch_request` method, you should be able to just return the result directly from `self.patch_request`

File: plugins/module_utils/redfish_utils.py (hunk 1 above)
Comment (TSKushal) [reply plugins/module_utils/redfish_utils.py]: Changed as suggested. Thanks!

File: plugins/module_utils/redfish_utils.py [hunk 3]
```diff
@@ -3152,3 +3152,53 @@ def set_session_service(self, sessions_config):
...
+        server_details = response["data"]
+        secure_boot_url = server_details["SecureBoot"]["@odata.id"]
+
+        response = self.get_request(self.root_uri + secure_boot_url)
+        if not response["ret"]:
+            return response
+
+        secure_boot_details = response["data"]
+
+        if secure_boot_details["SecureBootEnable"]:
```
[the remark this answers is not in the document]
Comment (TSKushal) [reply plugins/module_utils/redfish_utils.py]: Changed as suggested. Thanks!

File: plugins/module_utils/redfish_utils.py (hunk 2 above)
Comment (TSKushal) [reply plugins/module_utils/redfish_utils.py]: Changed as suggested. Thanks!

File: plugins/module_utils/redfish_utils.py [hunk 4]
```diff
@@ -3152,3 +3152,27 @@ def set_session_service(self, sessions_config):
...
+
+        body = {}
+        body["SecureBootEnable"] = True
+
+        response = self.patch_request(self.root_uri + secure_boot_url, body, check_pyld=True)
+
+        if response["ret"] is False:
+            return response
+
+        return response
```
Comment (mraineri) [inline plugins/module_utils/redfish_utils.py:3178]:
```suggestion
        return self.patch_request(self.root_uri + secure_boot_url, body, check_pyld=True)
```

File: plugins/modules/redfish_config.py [hunk 5]
```diff
@@ -124,7 +124,12 @@
     default: {}
     version_added: '5.7.0'
 
-author: "Jose Delarosa (@jose-delarosa)"
```
Comment (mraineri) [inline plugins/modules/redfish_config.py:127]: @felixfontein what's the normal practice for listing authors in a module? Is it just the original contributor or anyone actively developing?

Comment (felixfontein) [reply plugins/modules/redfish_config.py:127]: It depends a lot. Some folks never add themselves as authors, even though they basically rewrite the module, while others want to add themselves when implementing even smaller features. It's basically a matter of taste. I'm not sure that implementing an apparently simple feature like this one legitimates that four new authors should be added. Adding them to BOTMETA as maintainers is one thing, but to the `author` field looks really strange to me. @TSKushal can you please explain how all these authors contributed to this PR?

Comment (TSKushal) [reply plugins/modules/redfish_config.py:127]: We just thought this was the standard practice followed by everyone to add the author names for any contribution done.
All the authors mentioned here are actively contributing to the open source work of redfish modules from us. If it is inappropriate to add everyone  here for single method/small change, I can remove the others.

Comment (mraineri) [reply plugins/modules/redfish_config.py:127]: I'm fine with this. Maybe one day I'll add myself.

File: .github/BOTMETA.yml [hunk 6]
```diff
@@ -1036,7 +1036,7 @@ files:
   $modules/read_csv.py:
     maintainers: dagwieers
   $modules/redfish_:
-    ignore: jose-delarosa
+    ignore: jose-delarosa TSKushal
```
Comment (mraineri) [inline .github/BOTMETA.yml:1039]: Same comment as in #5900

File: plugins/module_utils/redfish_utils.py (hunk 1 above)
Comment (TSKushal) [reply plugins/module_utils/redfish_utils.py]: Resolving this conversation. Thanks!

File: plugins/module_utils/redfish_utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (TSKushal) [reply plugins/module_utils/redfish_utils.py]: Resolving this conversation. Thanks!

File: plugins/module_utils/redfish_utils.py (hunk 2 above)
Comment (TSKushal) [reply plugins/module_utils/redfish_utils.py]: Resolving this conversation. Thanks!

File: plugins/modules/redfish_config.py (hunk 5 above)
Comment (TSKushal) [reply plugins/modules/redfish_config.py:127]: Hope this is clarified now. Can I resolve it?

File: .github/BOTMETA.yml (hunk 6 above)
Comment (TSKushal) [reply .github/BOTMETA.yml:1039]: Hope this is clarified now. Can I resolve it?

Comment (felixfontein) [reply .github/BOTMETA.yml:1039]: See https://github.com/ansible-collections/community.general/pull/5900#discussion_r1103192899.

Comment (TSKushal) [reply .github/BOTMETA.yml]: Replied on [#5900](https://github.com/ansible-collections/community.general/pull/5900#discussion_r1103192899.)

**Review by russoz — COMMENTED** [review]
We are missing tests for this module, so it is hard to assess this change.

Other than that, it *looks* good to me, from the Ansible perspective.

**Comment by mraineri** [comment]
shipit

**Comment by TSKushal** [comment]
> Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

Done 😄

File: changelogs/fragments/5899-adding-enablesecureboot-functionality-to-redfish-config.yml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - redfish_command - adding EnableSecureBoot functionality (https://github.com/ansible-collections/community.general/pull/5899).
```
Comment (felixfontein) [inline changelogs/fragments/5899-adding-enablesecureboot-functionality-to-redfish-config.yml:2]:
```suggestion
  - redfish_command - adding ``EnableSecureBoot`` functionality (https://github.com/ansible-collections/community.general/pull/5899).
```

**Comment by felixfontein** [comment]
Unsurprisingly there's now a conflict after I merged #5900 (that PR looked slightly more complex, so I merged it first). Could you please rebase?

**Comment by TSKushal** [comment]
> Unsurprisingly there's now a conflict after I merged #5900 (that PR looked slightly more complex, so I merged it first). Could you please rebase?

Done

**Comment by felixfontein** [comment]
@TSKushal thanks for your contribution!
@mraineri @russoz thanks for reviewing!

**Comment by TSKushal** [comment]
Thank you 😄 @felixfontein @mraineri @russoz

**Pull request merged**